File: laocai-bot-spring-boot-starter/src/main/java/indi/dkx/laocai/bot/model/segment/OutgoingRecordSegment.java
package indi.dkx.laocai.bot.model.segment;

import java.util.Objects;

public final class OutgoingRecordSegment implements Segment {

    record Data(String uri) {
    }

    private final Data data;

    private OutgoingRecordSegment(Data data) {
        this.data = data;
    }

    public static OutgoingRecordSegment of(String uri) {
        return new OutgoingRecordSegment(new Data(Objects.requireNonNull(uri)));
    }

    @Override
    public Object getData() {
        return data;
    }

    @Override
    public String getType() {
        return "record";
    }
    
}

File: laocai-bot-spring-boot-starter/src/main/kotlin/indi/kyson/laocai/bot/segment/OutgoingRecordSegment.kt
package indi.kyson.laocai.bot.segment

data class OutgoingRecordSegment private constructor(override val data: Data) : Segment {

    data class Data(val uri: String) {
        fun toSegment(): OutgoingRecordSegment = OutgoingRecordSegment(this)
    }

    override val type: String
        get() = "record"

    companion object {
        @JvmStatic
        fun of(uri: String): OutgoingRecordSegment = OutgoingRecordSegment(Data(uri))
    }
}
